Preflight: add docs on oauth2Request (#430)
apollographql/docs: src/content/graphos/explorer/connecting-authenticating.mdx

<tr>
<td>

<tr>
<td>

##### `explorer.oauth2Request`

`(authUrl: string, queryParams?: Record<string, string>) => Promise<Record<string, string> | null>`

</td>
<td>

> Function that prompts the user to authenticate using your OAuth 2.0 provider specified by `authUrl` and returns the OAuth 2.0 query params in a promise. If the user cancels the prompt, the promise resolves to `null`.

Reviewer: ```suggestion
Function that prompts the user to authenticate using your OAuth 2.0 provider (specified by `authUrl`) and returns the OAuth 2.0 query params in a promise. If the user cancels the prompt, the promise resolves to `null`.
```

Reviewer: @mayakoneval I think we might need to differentiate more between the `queryParams` _parameter_ and the query params being supplied in the promise. Is `queryParams` just the set of parameters that would normally be appended to `authUrl` when initiating the OAuth flow?

Author: yes correct to all

Author: its like response params & request query params but they are both query params

> For OAuth 2.0 Authentication to work on Studio, your OAuth 2.0 provider must recognize https://studio.apollographql.com/explorer-oauth2 as a valid redirect url. You can do this in your OAuth 2.0 provider settings when you set up a new client or application.

Reviewer: ```suggestion
For OAuth 2.0 authentication to work with the Explorer, your OAuth 2.0 provider must recognize `https://studio.apollographql.com/explorer-oauth2` as a valid redirect URL. You can configure this in your OAuth 2.0 provider settings when you set up a new client or application.
```
